### PR TITLE
Search: use only bluge-based search

### PR DESCRIPTION
**Pull request by FZambia**
**What this PR does / why we need it**:

Here we replacing current search V2 implementation based on handmade slice of dashboards and frontend filtering with bluge-only approach.

Important thing from PR: 

* Since panels now indexed as a separate documents we do extra effort to clean them up upon dashboard change (ex. one panel removed) or dashboard delete (all panels removed).

TODO:

* [x] ~Proper locations on partial updates still TBD - looks like we need a database query to load folder UID.~
* [x] ~Need to rewrite existing tests if we proceed with this.~

### CODE REVIEW

**Comment by FZambia** [comment]
> looks like we need a database query to load folder UID.

Or just an index lookup

File: pkg/services/searchV2/bluge.go [hunk 1]
```diff
@@ -30,145 +30,212 @@ const (
...
-	// open an index writer using the configuration
-	writer, err := bluge.OpenWriter(config)
+func initIndex(dashboards []dashboard, logger log.Logger) (*bluge.Reader, *bluge.Writer, error) {
+	writer, err := bluge.OpenWriter(bluge.InMemoryOnlyConfig())
 	if err != nil {
-		return nil, fmt.Errorf("error opening writer: %v", err)
+		return nil, nil, fmt.Errorf("error opening writer: %v", err)
 	}
-	defer func() {
-		err = writer.Close()
```
Comment (FZambia) [inline pkg/services/searchV2/bluge.go:42]: spent a lot of time finding out why updates do not work - turned out due to closed writer (we did not use it before after first indexing)

**Comment by FZambia** [comment]
Added some tests, think we need at least several more – for updates.

**Comment by ryantxu** [comment]
@FZambia -- I updated the response so we do *not* return score (or max_score) unless we are explicitly debugging.  This makes sense because the absolute values are kinda nonsense, they depend mostly on the strategies we take and what the corpus looks like.  I expect the exact values to change frequently -- and end users do not care what they are are anyway.

This will also make the tests more stable :) 🎉

**Comment by ryantxu** [comment]
@FZambia -- I would like to merge this soon because it is helpful for the people testing/building frontend parts in parallel :)

**Review by ryantxu — APPROVED** [review]
looks great -- lets improve & extend in additional PRs

**Pull request merged**